syntax = "proto3";
package com.panelsw.ca;
option optimize_for = LITE_RUNTIME;
import "errorHandling.proto";

message ConsoleOuputRemap { 
	int32 from = 1;
	int32 to = 2;
	bytes regex = 3;
}

message ExecOnDetails{

	bytes command = 1;
	bytes obfuscatedcommand = 2;
	bytes workingdirectory = 3;
	
	map<int32, int32> exitCodeRemap = 4;
	map<string, bytes> environment = 5;

	bool async = 6;
	ErrorHandling errorHandling = 7;

	repeated ConsoleOuputRemap consoleOuputRemap = 8;
}
